device/common/include/traccc/finding/device/build_tracks.hpp

```diff
@@ -49,21 +49,14 @@ struct build_tracks_payload {
     vecmem::data::vector_view<const typename candidate_link::link_index_type>
         tips_view;
 
-    /**
-     * @brief View object to the vector of track candidates
-     */
-    track_candidate_container_types::view track_candidates_view;
+    vecmem::data::vector_view<const unsigned int> valid_tip_idx_view;
 
-    /**
-     * @brief View object to the vector of indices meeting the selection
-     * criteria
-     */
-    vecmem::data::vector_view<unsigned int> valid_indices_view;
+    const unsigned int* num_valid_tips;
```

Comment:
😕 If `valid_tip_idx_view` points at a resizable buffer, then why would you need a separate variable for the size?

Reply:
No real reason, array + pointer is just more clear and ergonomic in my opinion.